docs/pages/guides/navigation-migration.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Navigation Migration
+sidebar_icon: arrows-left-right
+---
+
+This guide explains how to migrate existing configurations that used `topNavigation`, `sidebar` and `customPages` to the new unified `navigation` configuration introduced in vNEXT.
+
+## Overview
+
+Navigation is now configured through a single `navigation` array. Items at the root level become top navigation tabs, while nested categories automatically form the sidebar. Custom pages are added using the `custom-page` item type.
+
+## Before and After
+
+```ts title="Before"
+const config: ZudokuConfig = {
+  topNavigation: [
+    { id: "docs", label: "Docs" },
+    { id: "api", label: "API" },
+  ],
+  sidebar: {
+    docs: [{ type: "doc", id: "introduction" }],
+  },
+  customPages: [{ path: "/playground", render: Playground, prose: false }],
+  apis: { type: "file", input: "./openapi.json", navigationId: "api" },
+};
+```
```

Comment:
We offer several transformers to highlight words/lines, maybe this can help to show changes?

https://zudoku.dev/docs/markdown/code-blocks#advanced-syntax-highlighting